src/ApparentHorizons/StrahlkorperGr.cpp

```diff
@@ -6,14 +6,345 @@
 #include <array>
 #include <cmath>
 #include <cstddef>
+#include <utility>
 
+#include "ApparentHorizons/SpherepackIterator.hpp"
 #include "ApparentHorizons/YlmSpherepack.hpp"
 #include "DataStructures/DataVector.hpp"
+#include "DataStructures/Matrix.hpp"
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
 #include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
 #include "DataStructures/Tensor/Tensor.hpp"
+#include "NumericalAlgorithms/LinearAlgebra/FindGeneralizedEigenvalues.hpp"
 #include "PointwiseFunctions/GeneralRelativity/IndexManipulation.hpp"
 #include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Gsl.hpp"
 #include "Utilities/MakeWithValue.hpp"
+template <typename X, typename Symm, typename IndexList>
+class Tensor;
+
+// Functions used by StrahlkorperGr::dimensionful_spin_magnitude
+namespace {
+// Find the surface metric by inserting tangents
+// into the slots of the spatial metric
+template <typename Fr>
+tnsr::ii<DataVector, 2, Frame::Logical> get_surface_metric(
+    const tnsr::ii<DataVector, 3, Fr>& spatial_metric,
+    const StrahlkorperTags::StrahlkorperTags_detail::Jacobian<Fr>& tangents,
+    const Scalar<DataVector>& sin_theta) noexcept {
+  auto surface_metric =
+      make_with_value<tnsr::ii<DataVector, 2, Frame::Logical>>(
+          spatial_metric.get(0, 0), 0.0);
+  for (size_t i = 0; i < 3; ++i) {
+    for (size_t j = 0; j < 3; ++j) {
```

Comment:
I tend to prefer simplicity here, unless this turns out to be a bottleneck demonstrated by profiling.

Reply:
OK, my next commit will include this optimization, since it saves 3 terms in the sum, but I think the above suggestion that saves one sum is not worth the extra complexity.